examples/tests/sdk_tests/bucket/signed_url.test.w

```diff
@@ -1,29 +1,72 @@
 bring cloud;
-bring http;
+bring fs;
 bring util;
+bring expect;
 
-let testBucket = new cloud.Bucket(public: true) as "testBucket";
+let bucket = new cloud.Bucket();
 
-test "signedUrl" {
-  let var error = "";
-  testBucket.put("file1.txt", "Foo");
-  
-  if (util.env("WING_TARGET") != "sim") { 
-    let signedUrl = testBucket.signedUrl("file1.txt");
-    assert(http.get(signedUrl).body ==  "Foo"); 
-  } 
+test "signedUrl GET" {
+  let KEY = "tempfile.txt";
+  let VALUE = "Hello, Wing!";
+
+  bucket.put(KEY, VALUE);
+
+  let getSignedUrl = bucket.signedUrl(KEY);
+
+  // Download file from private bucket using GET presigned URL
+  let output = util.shell("curl \"{getSignedUrl}\"");
+
+  expect.equal(output, VALUE);
 }
 
-test "signedUrl for non existent key" {
-  let var error = "";
-  
-  if (util.env("WING_TARGET") != "sim") { 
-    try{
-      let signedUrl = testBucket.signedUrl("file.txt");
-    } catch e {
-      error = e;
+test "signedUrl GET with unexisting key" {
+  let assertThrows = (expected: str, block: (): void) => {
+    let var error = false;
+    try {
+      block();
+    } catch actual {
+      expect.equal(actual, expected);
+      error = true;
     }
+      expect.equal(error, true);
+  };
+  let UNEXISTING_KEY = "no-such-file.txt";
+  let OBJECT_DOES_NOT_EXIST_ERROR = "Cannot provide signed url for a non-existent key (key={UNEXISTING_KEY})";
+
+  assertThrows(OBJECT_DOES_NOT_EXIST_ERROR, () => {
+    bucket.signedUrl(UNEXISTING_KEY);
+  });
+}
+
+test "signedUrl PUT" {
+  let KEY = "tempfile.txt";
+  let VALUE = "Hello, Wing!";
+
+  let tempDir = fs.mkdtemp();
+  let tempFile = fs.join(tempDir, KEY);
+  fs.writeFile(tempFile, VALUE);
```

Comment:
the tmp file appears in the PR, can you delete the tmp dir afterward?

Reply:
That was actually manually added by me at first. The `tempFile` automatically gets deleted as it's placed inside a `tempDir` (temp dirs are cleaned up at the end of tests as part of the test routine).